test/browser/fixture.test.js

```diff
@@ -31,6 +27,31 @@ const slowElem = defineCE(
 	}
 );
 
+const asyncElem = defineCE(
+	class extends LitElement {
+		constructor() {
+			super();
+			this.finished = false;
+			this.promise = new Promise((resolve) => resolves.set(this.id, resolve));
+			this.promise.then(() => this.finished = true);
+		}
+		render() {
+			return html`<slot></slot>`;
+		}
+		async getLoadingComplete() {
```

Comment:
This is kinda outside the scope of this PR, but worth bring up because I believe James might be starting on the skeleton group loading stuff soon - do you envision this being the same piece that could be used for the grouped skeletons, to tell the parent "I'm done loading and awaiting confirmation to turn off my skeleton". Or by all HM components in general, to indicate they're done?

Or do you see this as mostly a test/vdiff thing, and those will use cases will look more like the async container stuff?

Reply:
That's a good question, these things are definitely related.

The mechanism as it's used here is more of a 1-time thing that the tests wait for, whereas theoretically the skeleton stuff could go from loading -> ready and then back to loading, in which case the skeleton container and all the items in it should likely go back into skeleton mode.

For that reason my mind went more to the subscriber stuff for the skeletons. This also doesn't need to be set in stone -- once we figure out what works for the skeleton collection we can try this out with it and also try out visual-diff in some hypermedia repos and then adjust as necessary.